src/frequenz/sdk/timeseries/_base_types.py

```diff
@@ -147,6 +180,29 @@ class Bounds(Generic[_T]):
     upper: _T
     """Upper bound."""
 
+    def __contains__(self, item: _T) -> bool:
+        """
+        Check if the value is within the range of the container.
+
+        Args:
+            item: The value to check.
+
+        Returns:
+            bool: True if value is within the range, otherwise False.
+        """
+        if item is None:
+            return False
+
+        assert isinstance(item, Comparable)
```

Comment:
Now that I think about it, I think `item = typing.cast(Comparable, item)` would perform better because it doesn't do any runtime type checks, and is just a hint to the type checker.

Doing that would also make it clear that it is being done because the typechecker has trouble otherwise.  Here I had to think for a minute to remember why the assert is required.